girder_worker_utils/transforms/girder_io.py

```diff
@@ -37,36 +40,68 @@ def __init__(self, *args, **kwargs):
             self.gc = None
 
 
+@six.add_metaclass(abc.ABCMeta)
 class GirderClientResultTransform(ResultTransform, GirderClientTransform):
     pass
 
 
-class GirderFileId(GirderClientTransform):
+@six.add_metaclass(abc.ABCMeta)
+class _GirderResourceDownload(GirderClientTransform):
+    def __init__(self, _id, **kwargs):
+        super(_GirderResourceDownload, self).__init__(**kwargs)
+        self._id = _id
+        self._local_path = None
+
+    def _repr_model_(self):
+        return "{}('{}')".format(self.__class__.__name__, self._id)
+
+    @property
+    def local_path(self):
+        if self._local_path is None:
+            self._local_path = os.path.join(tempfile.mkdtemp(), str(self._id))
```

Comment:
In the case of an item with multiple files, the files will still retain their names from the server, and will be siblings on the filesystem: https://github.com/girder/girder/blob/master/clients/python/girder_client/__init__.py#L1294-L1309

However, I do think it's a bug that a single file download loses its name. Placing each resource inside a folder by its ID is a good feature because it prevents collision based on name, but we don't want to lose either the name or the file extension.

I think I'd prefer to fix that problem on a future PR, though.

Reply:
Is this something that we can let the calling context decide by adding an optional `file_name` kwarg to `__init__(...)` and just using the ID if its not passed?

Reply:
That was my plan.